BootloaderCommonPkg/Library/TpmLib/TpmEventLog.c

```diff
@@ -259,11 +259,20 @@ TpmLogSpecIDEvent (
   if ((ActivePcr & HASH_ALG_SHA256) != 0) {
     DigestSize[NumberOfAlgorithms].algorithmId = TPM_ALG_SHA256;
     DigestSize[NumberOfAlgorithms].digestSize = SHA256_DIGEST_SIZE;
-
-    DigestSize = &DigestSize[1];
-    ++NumberOfAlgorithms;
+  } else if ((ActivePcr & HASH_ALG_SHA384) != 0) {

```

Comment:
Let's replace 'else if' with 'if' here.
This will allow in future to support multiple PCR banks simultaneously.

Also, do NumberOfAlgorithms++ in every 'if' section.

Reply:
yes sachin.